modules/sc-mesh-secure-deployment/src/nats/src/cbma_adaptation.py

```diff
@@ -605,10 +599,13 @@ def __set_mtu_size(self, interface_name: str, interface_color: str = None) -> No
         self.logger.info("Setting MTU size for %s to %s", interface_name, mtu_size)
 
         try:
-            subprocess.run(
-                ["ip", "link", "set", "dev", interface_name, "mtu", mtu_size],
-                check=True,
+            ret = subprocess.run(
+                ["ip", "link", "set", "dev", interface_name, "mtu", mtu_size], check=True
```

Comment:
I believe the `check=True` shall be removed if the `returncode` will be checked below, otherwise an [CalledProcessError](https://docs.python.org/3.9/library/subprocess.html#subprocess.CalledProcessError) will be raised when the command returns non-zero :boom:

Reply:
[78001c8](https://github.com/tiiuae/mesh_com/pull/456/commits/78001c855c0500804ddeda6c0b3412b7a4207b9e)

Reply:
Ah, in this case, if you are only handling the CalledProcessError and are not interested in the command output you can use subprocess.check_call() instead of subprocess.run() 🧄